arwendebug/messagesBase.go

```diff
@@ -0,0 +1,80 @@
+package arwendebug
+
+import (
+	"math/big"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// RequestBase is a CLI / REST request message
+type RequestBase struct {
+	DatabasePath string
+	World        string
+	Outcome      string
+}
+
+func (request *RequestBase) digest() error {
+	if request.DatabasePath == "" {
+		request.DatabasePath = "./db"
+	}
+
+	if request.World == "" {
+		request.World = "default"
+	}
+
+	return nil
+}
+
+// ResponseBase is a CLI / REST response message
+type ResponseBase struct {
+	Error error
+}
+
+// ContractRequestBase is a CLI / REST request message
+type ContractRequestBase struct {
+	RequestBase
+	ImpersonatedHex string
+	Impersonated    []byte
+	Value           string
+	ValueAsBigInt   *big.Int
+	GasPrice        uint64
+	GasLimit        uint64
+}
+
+func (request *ContractRequestBase) digest() error {
+	err := request.RequestBase.digest()
+	if err != nil {
+		return err
+	}
+
+	if request.ImpersonatedHex == "" {
+		return NewRequestError("empty impersonated address")
+	}
+
+	request.Impersonated, err = fromHex(request.ImpersonatedHex)
+	if err != nil {
+		return NewRequestErrorMessageInner("invalid impersonated address", err)
+	}
+
+	if request.GasPrice == 0 {
+		request.GasPrice = DefaultGasPrice
+	}
+
+	if request.GasLimit == 0 {
+		request.GasLimit = DefaultGasLimit
```

Comment:
I wouldn't do this - make it mandatory to call to come with GasPrice and GasLimit. That is how they learn how to use.

Reply:
Fixed.